app/components/page_list_component/view.rb

```diff
@@ -19,5 +19,35 @@ def show_down_button(index)
     def question_text_for_page(id)
       @pages.find { |page| page.id == id }.question_text
     end
+
+    def error_id(number)
+      "condition_#{number}"
+    end
+
+    def error_link(error_name, edit_link)
+      content_tag(
+        :a,
+        I18n.t("page_conditions.errors.#{error_name}"),
+        class: "govuk-link app-page_list__route-text--error",
+        href: edit_link,
+      )
+    end
+
+    def answer_value_text_for_condition(condition, edit_link)
+      if condition.errors_include?("answer_value_doesnt_exist")
+        error_link("answer_value_doesnt_exist", edit_link)
+      else
+        t("page_conditions.condition_answer_value_text", answer_value: condition.answer_value)
+      end
+    end
+
+    def goto_page_text_for_condition(condition, edit_link)
+      if condition.errors_include?("goto_page_doesnt_exist")
+        error_key = condition.errors_include?("answer_value_doesnt_exist") ? "goto_page_doesnt_exist_and_nor_does_answer_value" : "goto_page_doesnt_exist"
```

Comment:
do we need all this logic? Just thinking that the api should be responsible for api validation errors and would return a locale key and we could use that and not have to make up new ones in forms-admin

Reply:
I think even if we were passing the full message from the API a lot of the logic would still be needed, since we'd still need to decide whether to render a link or a normal string for each part of the condition unlkess we start sending the HTML from the API. But I do think it's worth keeping the translations in `forms-admin` since it'll a) make sure all of our translations are in one place and easier to update by content designers, and b) make things easier if we do need to add support for Welsh or other languages in future.

Would be interested in knowing your thoughts on moving some of this logic into the view - as I understand it it's cheaper to test in the view.rb file, but I think it might possibly be marginally more readable in view.html.erb, so I'm open to suggestions.

Reply:
We are pushing back on this so this shouldn't be needed for the time being